### ISSUE TITLE
Federation entity union is empty during schema introspection

### ISSUE THREAD

**Issue by ggrise** [async-graphql/async-graphql#700]
## Expected Behavior
union _Entity = ... list of entities..

## Actual Behavior

union _Entity = empty

## Steps to Reproduce the Problem

Create a federated entity and do a schema introspection, union _Entity remain empty.

I believe the bug was introduced in v2.10.3

## Specifications

- Version: v2.10.3
- Platform:
- Subsystem:

**Comment by sunli829** [comment]
Can you give an example to reproduce this problem? 🙂

**Comment by ggrise** [comment]
Yes, sorry for not giving more details:

If we run `/async-graphql/examples/federation/federation-reviews` and execute an introspection query on it
```
query IntrospectionQuery { __schema { types { name,kind,possibleTypes { name }  }}}"
```

We should expect:

```
{
  "kind": "UNION",
  "name": "_Entity",
  "possibleTypes": [
    {
      "name": "Product"
    },
    {
      "name": "User"
    }
  ]
},
```

But we get the following instead 

```
{
  "kind": "UNION",
  "name": "_Entity",
  "possibleTypes": []
},
```

thanks!

**Comment by sunli829** [comment]
Because this service itself does not contain any entities.

You should do it like this:

https://github.com/async-graphql/examples/blob/8641a72c8ccb78e924e2550cd1fbfa2e62319174/federation/README.md

**Comment by ggrise** [comment]
You can repeat it with `federation-products` example:

full details:

Run the federation product service (no need for a gateway)

```
$ cd examples/federation/federation-products
$ cargo run 
```

Curl the service directly with an introspection query:

```
$ curl 'http://localhost:4002/graphql' --data-raw '{"operationName":"IntrospectionQuery","variables":{},"query":"query IntrospectionQuery { __schema { types { name,kind,possibleTypes { name }  }}}"}' |  jq  '.data.__schema.types[]|select(.name == "_Entity")'
{
  "name": "_Entity",
  "kind": "UNION",
  "possibleTypes": []
}
```

Notice the `possibleTypes` should be set to `Product` since it's a federated entity in `federation-products`

Since the bug was introduced in v2.10.3, if you checkout `v2.10.2` and run `federation-products` again:
```
$ git checkout v2.10.2
$ cd examples/federation-products
$ cargo run
```

You get the correct answer for `possibleTypes`

```
$ curl 'http://localhost:4002/graphql' --data-raw '{"operationName":"IntrospectionQuery","variables":{},"query":"query IntrospectionQuery { __schema { types { name,kind,possibleTypes { name }  }}}"}' |  jq  '.data.__schema.types[]|select(.name == "_Entity")'
{
  "kind": "UNION",
  "name": "_Entity",
  "possibleTypes": [
    {
      "name": "Product"
    }
  ]
}
```

I believe the bug was introduced here:
https://github.com/async-graphql/async-graphql/blob/4793c362c355f4ff1a2552636d9da818c273541b/src/model/type.rs#L146

`union_values` is used to list the `possibleTypes`, but unfortunately, we don't add the types into `union_values` when we defined `_Entity`. https://github.com/async-graphql/async-graphql/blob/0f1e5a19fdcbcd4a7bce22264318b9963217c0d4/src/registry/mod.rs#L506

I can make a PR if you like.

**Comment by Miaxos** [comment]
Oops, it seems I forgot something, it's my commit: https://github.com/async-graphql/async-graphql/commit/a3b77775161d5960b9f15fa8c0ef45c15998a54f

**Comment by sunli829** [comment]
I'm working on it.

**Comment by sunli829** [comment]
Fixed in 3.0
